components/queryString/QueryParamsForm.tsx

```diff
@@ -0,0 +1,149 @@
+import Delete from '@mui/icons-material/Delete';
+import Grid from '@mui/material/Grid';
+import TextField from '@mui/material/TextField';
+import React from 'react';
+
+export type QueryParams = {
+  param: string;
+  value: string;
+  key: string;
+}[];
+export const defaultParams = [{ param: '', value: '', key: 'initial' }];
+
+export interface QueryParamsFormProps {
+  queryParams: QueryParams;
+  setQueryParams: (_queryParams: QueryParams) => void;
+  setQueryString: (_queryString: string) => void;
+}
+
+export function getQueryString(
```

Comment:
Can we protect the URL here if there is one?

e.g. if I type `https://google.com/?q=test`, then add a new parameter from the inputs it wipes out the URL and I get `q=test&param=value`.

Reply:
Good idea. I've pushed a commit that solves this by checking if the existing string is a valid URL and then using the JS URL object to update the params. Downside is that it still wipes the URL if it isn't technically valid (i.e. if something like http:// is missing).
Another option would be to split the string at '?' and just keep the first part no matter what it is. I think that would probably work fine, but I'm not sure if there are any edge cases where that might cause problems? Let me know if you have any thoughts.

I'll do the translation stuff later today.

Reply:
@nosecreek I think splitting at '?' is a good route here, I did a little exploration and at looks like the first '?' is treated as the start of a query and any subsequent '?' is treated as part of the parameter or value depending on which side of the '=' it is on.

Based on that, I think something like this should cover any edge cases I can think of:

```js
const input = 'https://google.com/?q=test?&thing?=value'

const parts = input.split('?');
const url = parts[0];
const query = parts.slice(1).join('?');

console.log(url, query); // "https://google.com/" "?q=test?&thing?=value"
```

Also, thanks for adding a test case for the URL!

Reply:
Makes sense, I changed it to use split instead.

Reply:
We're close on this, but it doesn't quite handle having '?' in a parameter or value, e.g. `https://google.com/?q=test?&thing?=four`

![2022-10-26_16-00-34](https://user-images.githubusercontent.com/22881850/198154656-67847bbd-3070-474c-bb28-70ee454a910f.png)